File: JAVA/Linked list of strings forms a palindrome.java
class Solution {
    public boolean compute(Node root) {
        StringBuilder sb=new StringBuilder();
        while(root!=null)
        {
            sb.append(root.data);
            root=root.next;
        }
        if((sb.toString()).equals(sb.reverse().toString()))
            return true;
            
        return false;
    }
}

File: C++/Linked list of strings forms a palindrome.cpp
class Solution {
  public:
    bool compute(Node* head) {
        string name1;
        while(head!=NULL)
        {
            name1=name1+head->data;
            head=head->next;
        }
        int i=0;
        int size=name1.size()-1;
        while(i<name1.size()/2)
        {
          if(name1[i]!=name1[size-i])
          return false;
          i++;
        }
    return true;
    }
};
